pex/platforms.py

```diff
@@ -5,14 +5,82 @@
 
 from collections import namedtuple
 
-from pex.pep425tags import (
-    get_abbr_impl,
-    get_abi_tag,
-    get_impl_ver,
-    get_platform,
-    get_supported,
-    get_supported_for_any_abi
-)
+from pex.orderedset import OrderedSet
+from pex.pep425tags import get_abbr_impl, get_abi_tag, get_impl_ver, get_platform, get_supported
+
+
+def _gen_all_compatible_versions(version):
+  # We select major and minor here in the context of implementation version strings.
+  # These are typically two digit characters; eg "27" or "36", but they can be three digit
+  # characters in the case of pypy; eg "271". In the typical case the 1st digit represents the
+  # python major version and the 2nd digit it's minor version. In the pypy case, the 1st digit still
+  # represents the (hosting) python major version, the 2nd the pypy major version and the 3rd the
+  # pypy minor version. In both cases the last digit is the minor version and the python in question
+  # guarantees backwards compatibility of minor version bumps within the major version as per
+  # semver.
```

Comment:
The fact that cp37 supports `py3{0-6}` was surprising to me. I recommending link to PEP 425.

```suggestion
  # semver and PEP 425 (https://www.python.org/dev/peps/pep-0425/#id1).
```